From 3ec8a1683b051f9c644cc697f0cfb937bb05a702 Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Wed, 30 Jan 2019 19:18:13 +0530
Subject: [PATCH 064/235] sdk: arm64: dts: reduce usdpaa memory to 4K for
 LS1046/43

This patch reducing the USDPAA reseved memory to 4K.
In case USDPAA is to be used, 256MB needs to be reserved in the DTS file.

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts | 9 +++++++--
 arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts | 9 +++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
index f6e714d0d4f7..d512c214c1e8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
@@ -113,11 +113,16 @@
 		#size-cells = <2>;
 		ranges;
 
+		/* For legacy usdpaa based use-cases, update the size and
+		   alignment parameters. e.g. to allocate 256 MB memory:
+		   size = <0 0x10000000>;
+		   alignment = <0 0x10000000>;
+		*/
 		usdpaa_mem: usdpaa_mem {
 			compatible = "fsl,usdpaa-mem";
 			alloc-ranges = <0 0 0x10000 0>;
-			size = <0 0x10000000>;
-			alignment = <0 0x10000000>;
+			size = <0 0x1000>;
+			alignment = <0 0x1000>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
index 540289c3ab11..060ba922613d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
@@ -106,11 +106,16 @@
 		#size-cells = <2>;
 		ranges;
 
+		/* For legacy usdpaa based use-cases, update the size and
+		   alignment parameters. e.g. to allocate 256 MB memory:
+		   size = <0 0x10000000>;
+		   alignment = <0 0x10000000>;
+		*/
 		usdpaa_mem: usdpaa_mem {
 			compatible = "fsl,usdpaa-mem";
 			alloc-ranges = <0 0 0x10000 0>;
-			size = <0 0x10000000>;
-			alignment = <0 0x10000000>;
+			size = <0 0x1000>;
+			alignment = <0 0x1000>;
 		};
 	};
 };
-- 
2.17.1

